server/pulp/plugins/util/publish_step.py

```diff
@@ -69,8 +74,9 @@ def __init__(self, step_type, status_conduit=None, non_halting_exceptions=None):
         self.last_report_time = 0
         self.last_reported_state = self.state
         self.timestamp = str(time.time())
-        self.non_halting_exceptions = non_halting_exceptions
+        self.non_halting_exceptions = non_halting_exceptions or []
```

Comment:
I'll add the docstring.

Adding non_halting_exceptions=[] in the parameter def would mean that the template value would be muteable.